### PR TITLE
LPS-144997,LPS-144998

### PR DESCRIPTION
**Pull request by p-albuquerque**
Allow submit ObjectEntry through a form in some specific situations. 

Related issue: 
https://issues.liferay.com/browse/LPS-144997; 
https://issues.liferay.com/browse/LPS-144998.

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:reevaluate:1337401_1177

**Comment by mateussgn** [comment]
Hey, @p-albuquerque I think this is only a rebase problem.
Try to rebase your branch and force push it.

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
Fixed and refactored

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Review by mateussgn — APPROVED** [review]
LGTM! 🎉

File: modules/apps/object/object-dynamic-data-mapping/src/main/java/com/liferay/object/dynamic/data/mapping/internal/storage/ObjectDDMStorageAdapter.java [hunk 1]
```diff
@@ -376,7 +393,7 @@ private String _getOptionReferenceValue(
 		throws JSONException, ParseException {
 
 		DDMFormField ddmFormField = ddmFormFieldsMap.get(
-			ddmFormFieldValue.getName());
```
Comment (marcelabc) [inline modules/apps/object/object-dynamic-data-mapping/src/main/java/com/liferay/object/dynamic/data/mapping/internal/storage/ObjectDDMStorageAdapter.java:379]: Hey @p-albuquerque,
Why it was relevant to change the way of getting the ddmFormFieldsMap from the property fieldReference?

File: modules/apps/object/object-dynamic-data-mapping/src/main/java/com/liferay/object/dynamic/data/mapping/internal/storage/ObjectDDMStorageAdapter.java [hunk 2]
```diff
@@ -311,6 +313,21 @@ private long _getObjectDefinitionId(
...
+		Collection<DDMFormField> ddmFormFields = ddmFormFieldsMap.values();
+
+		ddmFormFields.forEach(
+			ddmFormField -> {
+				if (_ddmFormfieldTypesNotMappable.contains(
+						ddmFormField.getType())) {
+
+					String ddmFormFieldName = ddmFormField.getName();
+
+					ddmFormFieldValues.removeIf(
```
Comment (marcelabc) [inline modules/apps/object/object-dynamic-data-mapping/src/main/java/com/liferay/object/dynamic/data/mapping/internal/storage/ObjectDDMStorageAdapter.java:325]: I was thinking about doing this with a different approach. Could we talk later?

File: modules/apps/object/object-dynamic-data-mapping/src/main/java/com/liferay/object/dynamic/data/mapping/internal/storage/ObjectDDMStorageAdapter.java [hunk 3]
```diff
@@ -515,6 +532,10 @@ else if (Objects.equals(objectFieldDBType, "Long")) {
 	private static final Log _log = LogFactoryUtil.getLog(
 		ObjectDDMStorageAdapter.class);
 
+	private final Set<String> _ddmFormfieldTypesNotMappable = SetUtil.fromArray(
```
Comment (marcelabc) [inline modules/apps/object/object-dynamic-data-mapping/src/main/java/com/liferay/object/dynamic/data/mapping/internal/storage/ObjectDDMStorageAdapter.java:535]: I remembered that these field types are categorized as transient in the DDMForm context. Transient fields are fields that don't save values from the user. 
Saying that, we could rename this variable.

**Review by marcelabc — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
After talk about the points brought by Marcela. Some changes was requested and done.

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

Hey @p-albuquerque, 
if you tested for both bugs and it is working, you can forward this PR. Thank you 😄

**Comment by liferay-forms** [comment]
Hey @p-albuquerque, can you please check if these failures are related to your changes? Thanks

**Comment by p-albuquerque** [comment]
ci:reevaluate:1297401_2053

**Comment by liferay-forms** [comment]
Hey @p-albuquerque, can you please check if these failures are related to your changes? Thanks

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:forward

**Pull request closed without merging**